sdks/python/apache_beam/runners/dataflow/internal/apiclient_test.py

```diff
@@ -44,6 +44,33 @@ def test_create_application_client(self):
     pipeline_options = PipelineOptions()
     apiclient.DataflowApplicationClient(pipeline_options)
 
+  def test_pipeline_url(self):
+    pipeline_options = PipelineOptions(
+        ['--subnetwork', '/regions/MY/subnetworks/SUBNETWORK',
+         '--temp_location', 'gs://any-location/temp'])
+    env = apiclient.Environment([],
+                                pipeline_options,
+                                '2.0.0', # any environment version
+                                FAKE_PIPELINE_URL)
+
+    recovered_options = None
+    for additionalProperty in env.proto.sdkPipelineOptions.additionalProperties:
+      if additionalProperty.key == 'options':
+        recovered_options = additionalProperty.value
+        break
+    self.assertIsNotNone(pipeline_options,
```

Comment:
for loops can have an "else" clause that's triggered if no break was encountered.